neo/utils.py

```diff
@@ -0,0 +1,856 @@
+            else:
+                valid_mask = np.zeros(obj.shape)
+                if type(check_value) != str:
+                    warnings.warn(
+                        'Length of annotation "%s" (%s) does not fit '
+                        'to length of object list (%s)' % (
+                            annotation_key, len(check_value), len(obj)))
+
+        # extract object entries, which match requested annotation
+        else:
+            valid_mask = np.zeros(obj.shape)
+            for obj_id in range(len(obj)):
+                if check_value[obj_id] in annotation_value:
+                    valid_mask[obj_id] = True
+    else:
+        valid_mask = np.zeros(obj.shape)
+
+    valid_ids = np.where(valid_mask)[0]
+
+    return valid_ids
+
+
+def _get_valid_annotations(obj, valid_ids):
+    """
+    Internal function
+    """
+    sparse_annotations = copy.deepcopy(obj.annotations)
+    for key in sparse_annotations:
+        if _is_annotation_list(sparse_annotations[key], len(obj)):
+            sparse_annotations[key] = list(np.array(sparse_annotations[key])[
+                valid_ids])
+    return sparse_annotations
+
+
+def _is_annotation_list(value, exp_length):
+    """
+    Internal function
+    """
+    return (
+        (isinstance(value, list) or (
+            isinstance(value, np.ndarray) and value.ndim > 0)) and (len(value) == exp_length))
+
+
+def add_epoch(
+        segment, event1, event2=None, pre=0 * pq.s, post=0 * pq.s,
+        attach_result=True, **kwargs):
+    """
+    Create epochs around a single event, or between pairs of events. Starting
+    and end time of the epoch can be modified using pre and post as offsets
+    before the and after the event(s). Additional keywords will be directly
+    forwarded to the epoch intialization.
+
+    Parameters:
+    -----------
+    segment : neo.Segment
+        The segment in which the final Epoch object is added.
+    event1 : neo.Event
+        The Neo Event objects containing the start events of the epochs. If no
+        event2 is specified, these event1 also specifies the stop events, i.e.,
+        the epoch is cut around event1 times.
+    event2: neo.Event
+        The Neo Event objects containing the stop events of the epochs. If no
+        event2 is specified, event1 specifies the stop events, i.e., the epoch
+        is cut around event1 times. The number of events in event2 must match
+        that of event1.
+    pre, post: Quantity (time)
+        Time offsets to modify the start (pre) and end (post) of the resulting
+        epoch. Example: pre=-10*ms and post=+25*ms will cut from 10 ms before
+        event1 times to 25 ms after event2 times
+    attach_result: bool
+        If True, the resulting Neo Epoch object is added to segment.
+
+    Keyword Arguments:
+    ------------------
+    Passed to the Neo Epoch object.
+
+    Returns:
+    --------
+    epoch: neo.Epoch
+        An Epoch object with the calculated epochs (one per entry in event1).
```

Comment:
I suggest adding a section "See also", referring to the `Event.to_epoch()` method.

Reply:
Done.